src/runtime/internal/dom.ts

```diff
@@ -22,7 +22,7 @@ export function append(target: Node, node: Node) {
 	if (is_hydrating) {
 		nodes_to_detach.delete(node);
 	}
-	if (node.parentNode !== target) {
+	if (node.parentNode !== target || node.parentNode !== null) {
```

Comment:
hmm..reading from face value, this doesn't seems like a correct fix, there's maybe something wrong earlier in the hydration process / rather in the append method.

adding `|| node.parent !== null` seems to defeat the purpose of doing `node.parentNode !== target` check

Reply:
Yes, that would defeat the check and so I agree the code shouldn't be written this way. It's also worth noting though that the check was introduced along with the bug in https://github.com/sveltejs/svelte/pull/6204 so we may not need to keep the check as is. If you remove this check and the one in `insert` to go back to how the code was originally then the tests will still pass.

Reply:
the alternate of this is 

```js
if (node.parentNode !== target || node.nextSibling !== null) {
```

taking care the case where node parent is already correct, but the order is wrong

Reply:
In `insert` the following passes as well:
```ts
if (
	node.parentNode !== target ||
	(!anchor && node.nextSibling !== null) ||
	(anchor && node.nextSibling !== anchor)
) {
```
Definitely looks much more sensible than my initial attempt, although I don't know there still are other edge cases not caught by the tests. Would there be any good way to test the idempotence property more directly?